### PR TITLE
Docs improvements

### PR DESCRIPTION
**Pull request by nalimilan**
Include all exported docstrings in the index, fix links and doctests and require them to pass.

### CODE REVIEW

File: src/array.jl [hunk 1]
```diff
@@ -615,7 +615,7 @@ similar(::Type{T}, dims::Dims) where {U, T<:Array{Union{CategoricalValue{U}, Mis
     compress(A::CategoricalArray)
 
 Return a copy of categorical array `A` using the smallest reference type able to hold the
-number of [`levels`](@ref) of `A`.
+number of [`levels`](@ref DataAPI.levels) of `A`.
```
Comment (bkamins) [inline src/array.jl:618]: Does the link `DataAPI.levels` work?

Comment (nalimilan) [reply src/array.jl:618]: Yes it does! :-)

Comment (bkamins) [reply src/array.jl:618]: Wow - it has not some time ago! This is great.

Comment (bkamins) [reply src/array.jl:618]: I have just checked the docs. It prints "Missings.levels" as a function name. But this is minor.

**Pull request merged**